tools/themis/src/rules.rs

```diff
@@ -0,0 +1,351 @@
+use std::collections::HashMap;
+
+use super::{style, utils, Error, Expected, PackageOutcome, Workspace};
+
+/// ensure all crates have the `publish = <true/false>` specification
+pub fn has_publish_spec(workspace: &Workspace) -> Result<(), Error> {
+    let outliers: Vec<_> = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.raw["package"].get("publish").is_none())
+        .map(|pkg| PackageOutcome { pkg, value: None })
+        .collect();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages should have the `publish` specification".to_string(),
+            expected: Some(Expected { value: "publish = <true/false>".to_string(), reason: None }),
+            outliers,
+        });
+    }
+
+    return Ok(());
+}
+
+/// ensure all crates specify a MSRV
+pub fn has_rust_version(workspace: &Workspace) -> Result<(), Error> {
+    let outliers: Vec<_> = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.raw["package"].get("rust-version").is_none())
```

Comment:
No need to reach into `raw` here, this is exposed via cargo metadata.

Reply:
Also, yeah it seems like it should but apparently, while `rust-version` was introduced in v1.56, that field isn't going to be included into `cargo metadata` until v1.58.. which isn't even really out yet.